### PR TITLE
add new resources: continuous export and external table

### PR DESCRIPTION
**Pull request by FakieHeelflip**
- add continuous export as new resource
- add external table as new resource
- add documentation for both

Locally tested. Works as expected.

### CODE REVIEW

File: adx/resource_adx_external_table.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+			"data_format": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
+			},
+
+			"storage_connection_string": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
```
Comment (favoretti) [inline adx/resource_adx_external_table.go:58]: Can we mark this one as sensitive? I assume connection strings can contain secrets?
```suggestion
				ValidateDiagFunc: validate.StringIsNotEmpty,
				Sensitive: true,
```

File: adx/resource_adx_external_table.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+			},
+
+			"dry_run": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+		CustomizeDiff: clusterConfigCustomDiff,
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
```
Comment (favoretti) [inline adx/resource_adx_external_table.go:143]: Does this need to have a 30 *minutes* timeout? In certain cases TF will literally wait for 30 mins before failing.

File: adx/resource_adx_external_table.go [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+		d.SetId("")
+	} else {
+
+		d.Set("name", id.Name)
+		d.Set("database_name", id.DatabaseName)
+		d.Set("storage_connection_string", resultSet[0].ConnectionStrings)
+		d.Set("partitions", resultSet[0].Partitions)
+		d.Set("path_format", resultSet[0].PathFormat)
+		d.Set("folder", resultSet[0].Folder)
+		d.Set("properties", resultSet[0].Properties)
```
Comment (favoretti) [inline adx/resource_adx_external_table.go:250]: What about all other attributes? If those are set - wouldn't this generate a perpetual diff?

File: adx/resource_adx_table_continuous_export.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+	if diags.HasError() {
+		return diags
+	}
+
+	if len(resultSet) < 1 {
+		d.SetId("")
+	} else {
+
+		d.Set("name", id.Name)
+		d.Set("database_name", id.DatabaseName)
```
Comment (favoretti) [inline adx/resource_adx_table_continuous_export.go:184]: Same here re: reading other attributes as in previous resource.

File: adx/resource_adx_external_table.go (hunk 1 above)
Comment (FakieHeelflip) [reply adx/resource_adx_external_table.go:58]: Yes, that would be great. Gonna add this.

File: adx/resource_adx_external_table.go (hunk 2 above)
Comment (FakieHeelflip) [reply adx/resource_adx_external_table.go:143]: No, not really. Copy pasted it from another resource. What time do you suggest?

File: adx/resource_adx_external_table.go (hunk 3 above)
Comment (FakieHeelflip) [reply adx/resource_adx_external_table.go:250]: No, there is no perpetual diff. Honestly I can not figure out what we are doing with those read attributes. Other resources like materialized-views and functions also do not read all attributes they have as input parameters.

**Review by favoretti — APPROVED** [review]
LGTM, thank you

**Pull request merged**